writer/csv_writer.go

```diff
@@ -0,0 +1,50 @@
+package writer
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type CSVWriter struct {
+	file    *os.File
+	writer  *bufio.Writer
+	headers []string // To preserve field order
```

Comment:
nit: we can maintain a struct type for the csv headers and use that like how you have did for Parquet writer?

Reply:
I meant we can introduce a struct container for only the csv headers and use that here but NBD.